src/Application/QueryStringParser.php

```diff
@@ -38,11 +43,33 @@ private function splitQueryString( string $queryString ): array {
 		);
 	}
 
+	private function isInstancePart( string $part ): bool {
+		return str_starts_with( $part, 'haswbstatement:' );
```

Comment:
We need a new constructor argument: `private readonly PropertyId $instanceOfId`

Then in here we need to only match `'haswbstatement:' . $this->instanceOfId->getSerialization()`